src/network/networkd-dhcp-common.c

```diff
@@ -58,9 +58,16 @@ void network_adjust_dhcp(Network *network) {
         }
 }
 
+static struct DUID fallback_duid = { .type = DUID_TYPE_EN };
 DUID* link_get_duid(Link *link) {
         if (link->network->duid.type != _DUID_TYPE_INVALID)
                 return &link->network->duid;
+        else if (link->hw_addr.length == 0 &&
+                 (link->manager->duid.type == DUID_TYPE_LLT ||
+                  link->manager->duid.type == DUID_TYPE_LL))
```

Comment:
`IN_SET()` would be nicer here, but it doesn't matter much.

Reply:
Addressed in #17692.